### PR TITLE
Ability to use contextvars in logger

### PR DESCRIPTION
**Pull request by hh-h**
## What do these changes do?

As we discussed in issue below, I extracted and put some logic from `start` method into `_handle_request` method, but I don't like signature of the new method and a lot of tests are failed now. I tried to fix them, but no luck. Seems like we need to rethink the idea, but I need some directions.

## Are there changes in behavior for the user?

No

## Related issue number

https://github.com/aio-libs/aiohttp/issues/3787

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
The code change looks good.
Tests are failed because they are using mocking, but mocked functions are changed.
These tests should be fixed.

**Comment by webknjaz** [comment]
Hey @hh-h, could you please fix tests and add a change fragment?

**Comment by hh-h** [comment]
@webknjaz yes, I have this in my plans, but summer, you know :)

**Comment by webknjaz** [comment]
Right... It's just sometimes unclear whether the PR submitter gave up and ignores the PR if they don't post any information about their intentions.

File: aiohttp/web_protocol.py [hunk 1]
```diff
@@ -399,6 +400,32 @@ def _process_keepalive(self) -> None:
         self._keepalive_handle = self._loop.call_later(
             self.KEEPALIVE_RESCHEDULE_DELAY, self._process_keepalive)
 
+    async def _handle_request(self,
+                              request: BaseRequest,
+                              start: float) -> Tuple[StreamResponse, bool]:
```
Comment (webknjaz) [inline aiohttp/web_protocol.py]: Maybe?
```suggestion
                              start_time: float) -> Tuple[StreamResponse, bool]:
```

**Comment by hh-h** [comment]
@asvetlov @webknjaz PR updated

File: tests/test_web_log.py [hunk 2]
```diff
@@ -214,3 +216,33 @@ class Logger(AbstractAsyncAccessLogger):
     resp = await cli.get('/path/to', headers={'Accept': 'text/html'})
     assert resp.status == 200
     assert msg == '/path/to: 200'
+
+
+@pytest.mark.skipif(not PY_37,
+                    reason="contextvars support is required")
+async def test_contextvars_logger(aiohttp_server, aiohttp_client):
+    from contextvars import ContextVar
```
Comment (webknjaz) [inline tests/test_web_log.py]: plz put this at the top of this test module using `try/import/except ImportError` style

File: tests/test_web_protocol.py [hunk 3]
```diff
@@ -835,4 +835,4 @@ def test_data_received_force_close(srv) -> None:
     writer.write(b"x")
     writer.close()
     await asyncio.sleep(0.1)
-    logger.debug.assert_called_with('Ignored premature client disconnection 2')
+    logger.debug.assert_called_with('Ignored premature client disconnection.')
```
Comment (webknjaz) [inline tests/test_web_protocol.py:838]: How is this related to the contextvars change?

File: tests/test_web_server.py [hunk 4]
```diff
@@ -89,7 +89,7 @@
     with pytest.raises(client.ClientPayloadError):
         await resp.read()
 
-    logger.debug.assert_called_with('Ignored premature client disconnection ')
+    logger.debug.assert_called_with('Ignored premature client disconnection')
```
Comment (webknjaz) [inline tests/test_web_server.py:92]: unrelated

File: tests/test_web_protocol.py [hunk 5]
```diff
@@ -362,7 +362,7 @@ def close():
         b'GET / HTTP/1.1\r\n'
         b'Host: example.com\r\n'
         b'Content-Length: 0\r\n\r\n')
-    await asyncio.sleep(0)
+    await asyncio.sleep(0.01)
```
Comment (webknjaz) [inline tests/test_web_protocol.py:365]: Doesn't seem related.

File: tests/test_web_protocol.py (hunk 3 above)
Comment (hh-h) [reply tests/test_web_protocol.py:838]: This fixed tests, because I extracted some logic into `_handle_request`

File: tests/test_web_server.py (hunk 4 above)
Comment (hh-h) [reply tests/test_web_server.py:92]: see my comment above

File: tests/test_web_protocol.py (hunk 5 above)
Comment (hh-h) [reply tests/test_web_protocol.py:365]: Otherwise some tests are failing, my guess it's because I had to await one more task, so 0 is not enough.

File: tests/test_web_log.py (hunk 2 above)
Comment (hh-h) [reply tests/test_web_log.py]: will do

File: tests/test_web_protocol.py (hunk 3 above)
Comment (hh-h) [reply tests/test_web_protocol.py:838]: see https://travis-ci.com/aio-libs/aiohttp/jobs/213717795#L3044

**Comment by asvetlov** [comment]
Thanks!

**Comment by asvetlov** [comment]
This pull request has been mentioned on **aio-libs**. There might be relevant details there:

https://aio-libs.discourse.group/t/aiohttp-new-release/49/1
